docs/en/resource/supplemental.md

```diff
@@ -0,0 +1,351 @@
+
+	return (
+		<virtual>
+			// this List will be given the underlying resource
+			// with no shared `Options
+			<List resource={resource} />
+			// this List will share `Options` with the
+			// parent widget
+			<List resource={shared()} />
+		</virtual>
+	)
+}
+```
+
+This shared approach is used within our typeahead widget to share the query between the text input and the list of options. See more information on this in the `sharing` section below.
+
+## The Transform Property
+
+The `transform` property is used to convert the data returned by the `read` function to the format expected by the widget. When a widget uses the standard `data` middleware, `transform` is an optional property. However, when a widget creates a typed middleware using the `createDataMiddleware` function, it becomes a required property and is typed via a generic to ensure that the correct data format is created by the transform config.
+
+The transform is transparent and occurs in the middleware before data is returned to the widget. When performing queries on the resource, a reverse transform is used to convert the widget values back to the source values provided.
+
+```ts
+// resource data
+const resourceData = [{ firstname: 'joe', lastname: 'bloggs' }, { firstname: 'jane', lastname: 'doe' }];
+
+// with a transform of
+const transform = {
+	value: ['firstname']
+};
+
+// would create
+const transformedData = [{ value: 'joe' }, { value: 'jane' }];
+```
+
+Transforms can also be used to join source keys together to create composite fields. For example, with the same resource data as above
+
+```ts
+// with a transform of
+const transform = {
+	value: ['firstname', 'lastname']
+};
+
+// would create
+const transformedData = [{ value: 'joe bloggs' }, { value: 'jane doe' }];
+```
+
+### using a typed transform
+
+```tsx
+import { create } from '@dojo/framework/core/vdom';
+import { createDataMiddleware } from '@dojo/framework/core/middleware/data';
+
+// create the middleware with a generic
+const data = createDataMiddleware<{ value: string }>();
+const factory = create({ data });
+
+export const DataAwareWidget = factory(function ({
+	middleware: { data }
+}) {
+	const { getOrRead, getOptions } = data();
+	const data = getOrRead(getOptions());
+
+	return (
+		<virtual>
+			{ data.map(item => <span>{item.value}</span>) }
+		</virtual>;
+	)
+}
+
+// in render function elsewhere
+<List resource={resource} transform={{ value: ['firstname', 'lastname']}}>
+// if a key other than `value` was passed in the transform or no transform at all
+// was passed, a type error would occur
+```
+
+## API
+
+The data middleware API provides the widget with access to the resource data.
+
+### get options
+
+`getOptions` returns the current `Options` object. The result of this function should be passed to each of the `get` api functions. It is important to use `getOptions` rather than constructing a new `Options` object in order to ensure that shared resources etc work as expected.
```

Comment:
```suggestion
`getOptions` returns the current `Options` object. The result of this function should be passed to each of the `get` api functions. It is important to use `getOptions` rather than constructing a new `Options` object in order to ensure that shared resources work as expected.
```